### ISSUE TITLE
no explanation upon teleport into solid rock

### ISSUE THREAD

**Issue by hymie0** [atsb/RL_M#10]
It took me a few minute to figure out what happened.

I tried the Z command (mostly to see what it would do), and it took me right to the scoreboard and the game ended.  It looks like I "teleported into solid rock."

It would be nice if the game gave me some kind of message that this happened, prior to clearing the screen and ending the game.

**Comment by atsb** [comment]
I could add a small 5 second pause when this happens, I'll start working on it.

**Comment by atsb** [comment]
lprcat("You died by teleporting into solid rock.")

This was added into object.c.  Please re-test and see if it works.

**Comment by hymie0** [comment]
object.c: In function ‘oteleport’:
object.c:502: error: expected ‘;’ before ‘died’


(Thank you, by the way, for naming a weapon after me.  I'm touched.)

**Comment by hymie0** [comment]
object.c  Line 499-ish

```
void
oteleport (int err)
{
  int tmp;

  if (err)
    if (rnd (151) < 3)
    lprcat("You died by teleporting into solid rock.");
      died (264);   /* stuck in a rock */
  cdesc[TELEFLAG] = 1;    /* show ?? on bottomline if been teleported    */
  if (level == 0)
    tmp = 0;
```

You need braces around 
```
    lprcat("You died by teleporting into solid rock.");
      died (264);   /* stuck in a rock */
```
otherwise, the if (err) only has one line which is if(rnd) which only has one line which is lprcat.  When the if (err) fails, the next line is died(264).  So every scroll of teleport is an instant death.

**Comment by atsb** [comment]
I found this out when I was testing this last night.  I'm not sure why this if block was written without parentheses (unless it's really been untouched since 86).  Nice spotting though.

I did completely rewrite the oteleport function to handle user input for level choice and different things for volcano and caves, but ultimately went back to this approach because it changed things too much.  I'll add your fix to my trunk.

I wish other people had your energy and ability.
